transport/http2_server.go

```diff
@@ -143,6 +146,10 @@ func (t *http2Server) operateHeaders(hDec *hpackDecoder, s *Stream, frame header
 			hDec.state = decodeState{}
 		}
 	}()
+	if goaway {
+		//Stop creating streams on this transport
+		return nil
+	}
```

Comment:
we still need to process the headers frame to keep HPACK state consistent with the peer.